### PR TITLE
Address the Chapter 8 editorial review

### PR DESCRIPTION
**Pull request by jni**

### CODE REVIEW

File: markdown/ch8.markdown [hunk 1]
```diff
@@ -1,6 +1,18 @@
 # Big Data in Little Laptop with Toolz
 
-Whenever I think too hard about streaming data analysis, my head hurts.
+> GRACIE: A knife? The guy's twelve feet tall!
+> JACK: Seven. Hey, don't worry, I think I can handle him.
+>
+> — Jack Burton, Big Trouble in Little China
+
+Although streaming is not really a SciPy feature, it is a way of writing code
```
Comment (stefanv) [inline markdown/ch8.markdown]: Suggested rewording to try and capture:
- Contrast between this not being a SciPy feature but why we are still talking about it
- Emphasize that it is not a way of writing code, but a way to deconstruct the problem

We dedicate this chapter to streaming computation which, while not a SciPy feature as such, is an important way of computing solutions on large datasets using resource constrained machines (such as laptops).

File: markdown/ch8.markdown [hunk 2]
```diff
@@ -1,6 +1,18 @@
...
+that is critical to efficiently processing the large datasets that we see in
+science. The Python language contains some very nice primitives for streaming
```
Comment (stefanv) [inline markdown/ch8.markdown]: nice -> useful ?

gorgeous is a bit too much, perhaps

File: markdown/ch8.markdown [hunk 3]
```diff
@@ -18,40 +30,42 @@ print(sum_of_means)
 This strategy works really well for cases where your problem can be neatly solved with by-row processing.
 But things can quickly get out of hand when your code becomes more sophisticated.
 
-In traditional programming models, you pass your data to a function, the function processes the data, and then returns the result.
-Done.
+In the most commonly-used programming models, you pass your data to a function,
```
Comment (stefanv) [inline markdown/ch8.markdown]: commonly used

File: markdown/ch8.markdown [hunk 4]
```diff
@@ -18,40 +30,42 @@ print(sum_of_means)
...
-It's clear that for any kind of "big data" operation, this approach won't work.
+For any kind of "big data" operation, this approach won't work.
 
 Python's creators knew this, and created the "yield" keyword, which enables a function to process just one "sip" of the data, pass the result on to the next process, and *let the chain of processing complete* for that one piece of data before moving on to the next one.
 "Yield" is a rather nice name for it: the function *yields* control to the next function, waiting to resume processing the data until all the downstream steps have processed that data point.
 
-As I mentioned above, trying to think too hard about the flow of control in this paradigm is a surefire way to experience headaches, nausea, and other side effects.
+## Streaming with `yield`
+
 An awesome feature of Python is that it abstracts this complexity away, allowing you to focus on the analysis functionality.
```
Comment (stefanv) [inline markdown/ch8.markdown]: This sentence doesn't make quite as much sense with the previous removed.

File: markdown/ch8.markdown [hunk 5]
```diff
@@ -389,25 +399,34 @@ counts = tz.pipe('data/sample.fasta', open, c.filter(is_sequence),
...
 Another word for currying is partial evaluation.
 In functional programming, currying is a way to produce a function that can wait for the rest of the arguments to show up later.
 
-Currying is not named after the spice blend (though it does spice up your code).
-It is named for Haskell Curry, the mathematician who invented the concept.
-Haskell Curry is also the namesake of the Haskell programming language, which has functions curried by default!
+So, while the function call `map(np.log, numbers_list)` applies the `np.log`
+function to all of the numbers in `numbers_list` (returning a sequence of the
+logged numbers), the call `toolz.curried.map(np.log)` returns a *function* that
+takes in a sequence of numbers and returns a sequence of logged numbers.
```
Comment (stefanv) [inline markdown/ch8.markdown]: Clearly written!  I like it.

File: markdown/ch8.markdown (hunk 1 above)
Comment (jni) [inline markdown/ch8.markdown]: How about this mix? I want to stress the "science" angle to justify talking about it in a SciPy book. =)

> Although streaming is not a SciPy feature per se, it is a way of processing data
> that is critical to efficiently processing the large datasets that we see in
> science.

I also don't like "resource constrained machines" — every machine is resource constrained given a large enough dataset...

File: markdown/ch8.markdown (hunk 2 above)
Comment (jni) [inline markdown/ch8.markdown]: yep, fixed. =)

File: markdown/ch8.markdown (hunk 1 above)
Comment (stefanv) [inline markdown/ch8.markdown]: The "that is critical to efficiently processing" part doesn't parse well for me.

**Comment by stefanv** [comment]
All good apart from that one little sentence; if you'll humor me and shake it around a bit, I'm happy to merge.

**Pull request merged**